File: esports-backend/src/main/java/com/esports/service/EmailService.java
package com.esports.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.mail.SimpleMailMessage;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.stereotype.Service;

@Service
public class EmailService {

    @Autowired
    private JavaMailSender mailSender;

    public void sendOTP(String to, String otp) {
        try {
            SimpleMailMessage email = new SimpleMailMessage();
            email.setTo(to);
            email.setSubject("Upkraft - Email Verification OTP");
            email.setText("Your OTP for email verification is: " + otp + "\nThis OTP will expire in 10 minutes.");
            email.setFrom("upkraft.connect@gmail.com");
            
            mailSender.send(email);
        } catch (Exception e) {
            throw new RuntimeException("Failed to send OTP email: " + e.getMessage());
        }
    }

    public void sendProjectNotification(String to, String subject, String message) {
        try {
            SimpleMailMessage email = new SimpleMailMessage();
            email.setTo(to);
            email.setSubject(subject);
            email.setText(message);
            email.setFrom("upkraft.connect@gmail.com");
            
            mailSender.send(email);
        } catch (Exception e) {
            System.err.println("Failed to send notification email: " + e.getMessage());
        }
    }
}

File: esports-backend-dotnet/Upkart1/Services/EmailService.cs
using System.Net;
using System.Net.Mail;

namespace Upkart1.Services
{
    public class EmailService : IEmailService
    {
        private readonly IConfiguration _config;

        public EmailService(IConfiguration config)
        {
            _config = config;
        }

        public void SendOTP(string to, string otp)
        {
            try
            {
                var smtp = new SmtpClient(
                    _config["Email:Host"],
                    int.Parse(_config["Email:Port"]))
                {
                    Credentials = new NetworkCredential(
                        _config["Email:Username"],
                        _config["Email:Password"]
                    ),
                    EnableSsl = bool.Parse(_config["Email:EnableSsl"])
                };

                var mail = new MailMessage
                {
                    From = new MailAddress("upkraft.connect@gmail.com"),
                    Subject = "Upkraft - Email Verification OTP",
                    Body = $"Your OTP for email verification is: {otp}\nThis OTP will expire in 10 minutes.",
                    IsBodyHtml = false
                };

                mail.To.Add(to);
                smtp.Send(mail);
            }
            catch (Exception e)
            {
                throw new Exception($"Failed to send OTP email: {e.Message}");
            }
        }

        public void SendProjectNotification(string to, string subject, string message)
        {
            try
            {
                var smtp = new SmtpClient(
                    _config["Email:Host"],
                    int.Parse(_config["Email:Port"]))
                {
                    Credentials = new NetworkCredential(
                        _config["Email:Username"],
                        _config["Email:Password"]
                    ),
                    EnableSsl = bool.Parse(_config["Email:EnableSsl"])
                };

                var mail = new MailMessage
                {
                    From = new MailAddress("upkraft.connect@gmail.com"),
                    Subject = subject,
                    Body = message,
                    IsBodyHtml = false
                };

                mail.To.Add(to);
                smtp.Send(mail);
            }
            catch (Exception e)
            {
                Console.WriteLine($"Failed to send notification email: {e.Message}");
            }
        }

        public async Task SendProjectNotificationAsync(string to, string subject, string message)
        {
            try
            {
                var smtp = new SmtpClient(
                    _config["Email:Host"],
                    int.Parse(_config["Email:Port"]))
                {
                    Credentials = new NetworkCredential(
                        _config["Email:Username"],
                        _config["Email:Password"]
                    ),
                    EnableSsl = bool.Parse(_config["Email:EnableSsl"])
                };

                var mail = new MailMessage
                {
                    From = new MailAddress("upkraft.connect@gmail.com"),
                    Subject = subject,
                    Body = message,
                    IsBodyHtml = false
                };

                mail.To.Add(to);
                await smtp.SendMailAsync(mail);
            }
            catch (Exception e)
            {
                Console.WriteLine($"Failed to send notification email: {e.Message}");
            }
        }
    }
}
